pipecat-client-android-openai-realtime-webrtc/src/main/java/ai/pipecat/client/openai_realtime_webrtc/OpenAIRealtimeWebRTCTransport.kt

```diff
@@ -0,0 +1,429 @@
+                    JSON.encodeToJsonElement(
+                        OpenAIRealtimeSessionConfig.serializer(),
+                        initialConfig
+                    )
+                )))
+            }
+
+            if (initialUserMessage != null) {
+                options.add(Option(OPTION_INITIAL_USER_MESSAGE, Value.Str(initialUserMessage)))
+            }
+
+            return listOf(ServiceConfig(SERVICE_LLM, options))
+        }
+    }
+
+    object AudioDevices {
+        val Speakerphone = MediaDeviceInfo(
+            id = MediaDeviceId("speakerphone"),
+            name = "Speakerphone"
+        )
+
+        val Earpiece = MediaDeviceInfo(
+            id = MediaDeviceId("earpiece"),
+            name = "Earpiece"
+        )
+    }
+
+    class Factory(private val androidContext: Context) : TransportFactory {
+        override fun createTransport(context: TransportContext): Transport {
+            return OpenAIRealtimeWebRTCTransport(context, androidContext)
+        }
+    }
+
+    private var state = TransportState.Disconnected
+
+    private val appContext = androidContext.applicationContext
+    private val thread = transportContext.thread
+
+    private var client: WebRTCClient? = null
+
+    private val eventHandler = { msg: OpenAIEvent ->
+
+        thread.runOnThread {
+            when (msg.type) {
+                "error" -> {
+                    if (msg.error != null) {
+                        transportContext.callbacks.onBackendError(msg.error.describe() ?: "<null>")
+                    }
+                }
+
+                "session.created" -> {
+                    onSessionCreated()
+                }
+
+                "input_audio_buffer.speech_started" -> {
+                    transportContext.callbacks.onUserStartedSpeaking()
+                }
+
+                "input_audio_buffer.speech_stopped" -> {
+                    transportContext.callbacks.onUserStoppedSpeaking()
+                }
+
+                "response.audio_transcript.delta" -> {
+                    if (msg.delta != null) {
+                        transportContext.callbacks.onBotTTSText(
+                            MsgServerToClient.Data.BotTTSTextData(
+                                msg.delta
+                            )
+                        )
+                    }
+                }
+
+                "output_audio_buffer.started" -> {
+                    transportContext.callbacks.onBotStartedSpeaking()
+                }
+
+                "output_audio_buffer.cleared", "output_audio_buffer.stopped" -> {
+                    transportContext.callbacks.onBotStoppedSpeaking()
+                }
+
+                else -> {
+                    Log.i(TAG, "Ignoring incoming event with type '${msg.type}'")
+                }
+            }
+        }
+    }
+
+    override fun initDevices(): Future<Unit, RTVIError> = resolvedPromiseOk(thread, Unit)
```

Comment:
Are we going to implement it to initialize devices state and report initial available & selected devices ?